src/content/concepts/module-federation.md

```diff
@@ -0,0 +1,95 @@
+
+Loading remote modules is considered as async operation. When using a remote module these async operation will be placed in the next chunk loading operation(s) that is between the remote module and the entrypoint. It's not possible to use a remote module without a chunk loading operation.
+
+A chunk loading operation is usually an `import()`, but older constructs like `require.ensure` or `require([...])` are possible as well.
+
+A container is created through a container entry, which exposes async access to specified modules. The exposed access is separated into two steps:
+
+1. loading the module (async)
+2. evaluating the module (sync).
+
+Step 1 will be done during chunk loading. Step 2 will be done during module evaluation interleaved with other (local and remote) modules. This way evaluation order is unaffected by converting a module from local to remote or the other way around.
+
+Nesting container is possible. Containers can use modules from other containers. Circular dependencies between container are also possible.
+
+### Overriding
+
+A container is able to flag selected local modules as "overridable". A consumer of the container is able to provide "overrides", which are modules that replace one of the overridable module of the container. All modules of the container will use the replacement module instead of the local module when the consumer provides one. When the consumer doesn't provide a replacement module, all modules of the container will use the local one.
+
+The container will organize overridable modules in a way that they do not need to be downloaded when they has been overriding by the consumer. This usually happens by placing them into separate chunks.
+
+One the other hand the provider of the replacement modules will only provide async loading functions which allow the container to load replacement modules only when they are needed. The provider will organize replacement modules in a way that they do not need to be downloaded when they are not requested by the container at all. This usually happens by placing them into separate chunks.
+
+A "name" is used to identify overridable modules from the container.
+
+Overrides are provided in a similar way as the container exposes modules, separated into two steps:
+
+1. Loading (async)
+2. evaluating (sync).
+
+In nested scenarios, where a container uses other containers to load modules from, is overriding a transitive operation. Providing overrides to one container will also override these modules in nested container. This means that using a container will automatically make all "names" of the overridable modules, part of the overriding interface of the parent container.
+
+Overrides must be provided before modules of the container are loaded, otherwise behavior is undefined.
```

Comment:
Although, what do you mean "behaviour is undefined"
```suggestion
Overrides must be provided before the modules of the container are loaded, otherwise behavior is undefined.
```